lib/ansible/modules/cloud/oneandone/oneandone_user.py

```diff
@@ -0,0 +1,424 @@
+    oneandone_conn: authenticated oneandone object
+    """
+    user_id = module.params.get('name')
+    _user = get_user(oneandone_conn, user_id)
+
+    try:
+        user = oneandone_conn.delete_user(_user)
+
+        changed = True if user else False
+
+        return (changed, {
+            'id': user['id'],
+            'name': user['name']
+        })
+    except Exception as ex:
+        module.fail_json(msg=str(ex))
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            auth_token=dict(
+                type='str',
+                default=os.environ.get('ONEANDONE_AUTH_TOKEN')),
+            api_url=dict(
+                type='str',
+                default=os.environ.get('ONEANDONE_API_URL')),
+            name=dict(type='str'),
+            user=dict(type='str'),
+            description=dict(type='str'),
+            password=dict(type='str'),
+            email=dict(type='str'),
+            active=dict(type='bool'),
+            user_ips=dict(type='list', default=[]),
+            remove_ip=dict(type='str'),
+            change_api_key=dict(type='bool', default=False),
+            user_state=dict(
+                choices=USER_STATES),
+            wait=dict(type='bool', default=True),
+            wait_timeout=dict(type='int', default=600),
+            wait_interval=dict(type='int', default=5),
+            state=dict(type='str', default='present', choices=['present', 'absent', 'update']),
+        )
+    )
+
+    if not HAS_ONEANDONE_SDK:
+        module.fail_json(msg='1and1 required for this module')
+
+    if not module.params.get('auth_token'):
+        module.fail_json(
+            msg='auth_token parameter is required.')
+
+    if not module.params.get('api_url'):
+        oneandone_conn = oneandone.client.OneAndOneService(
+            api_token=module.params.get('auth_token'))
+    else:
+        oneandone_conn = oneandone.client.OneAndOneService(
+            api_token=module.params.get('auth_token'), api_url=module.params.get('api_url'))
+
+    state = module.params.get('state')
+
+    if state == 'absent':
+        if not module.params.get('name'):
+            module.fail_json(
+                msg="'name' parameter is required to delete a user.")
+        try:
+            (changed, user) = remove_user(module, oneandone_conn)
+        except Exception as ex:
+            module.fail_json(msg=str(ex))
+    elif state == 'update':
+        if not module.params.get('user'):
+            module.fail_json(
+                msg="'user' parameter is required to update a user.")
+        try:
+            (changed, user) = update_user(module, oneandone_conn)
+        except Exception as ex:
+            module.fail_json(msg=str(ex))
+
+    elif state == 'present':
+        for param in ('name', 'password'):
+            if not module.params.get(param):
+                module.fail_json(
+                    msg="%s parameter is required for new users." % param)
```

Comment:
not a blocker but there is a helper in the basic.py in  AnsibleModule `required_if` see https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/cloud/cloudstack/cs_vpc.py#L336 for an example.